dkg/dkg.go

```diff
@@ -26,30 +26,28 @@ type Actor interface {
 
 	// Encrypt encrypts the given message into kyber points
 	// using the DKG public key
-	Encrypt(message []byte) (K, C kyber.Point, remainder []byte, err error)
-	// Decrypt decrypts a pair of kyber points into the original message
-	// using the DKG internal private key
-	Decrypt(K, C kyber.Point) ([]byte, error)
+	// where K is the ephemeral DH (Diffie-Hellman) public key
+	// and Cs is the resulting, encrypted message
+	Encrypt(message []byte) (K kyber.Point, Cs []kyber.Point, err error)
+
+	// Decrypt decrypts a ciphertext (composed of a K and an array of C's)
+	// into the original message using the DKG internal private key
+	// where K is the ephemeral DH (Diffie-Hellman) public key
+	// and Cs is the encrypted message
+	Decrypt(K kyber.Point, Cs []kyber.Point) ([]byte, error)
+
+	// Reencrypt reencrypts generate a temporary key from the public key
+	// to be able to decrypt the message by the user's private key
+	Reencrypt(K kyber.Point, PK kyber.Point) (XhatEnc kyber.Point, err error)
 
 	// Reshare recreates the DKG with an updated list of participants.
 	Reshare(co crypto.CollectiveAuthority, newThreshold int) error
 
-	// VerifiableEncrypt encrypts the given message into kyber points
-	// using the DKG public key and proof of work algorithm.
+	// VerifiableEncrypt encrypts the given message into a ciphertext
+	// using the DKG public key and a verifiable encryption function.
 	VerifiableEncrypt(message []byte, GBar kyber.Point) (ciphertext types.Ciphertext, remainder []byte, err error)
 
 	// VerifiableDecrypt decrypts a pair of kyber points into the original message
-	// using the DKG internal private key and a proof of work algorithm.
+	// using the DKG internal private key and a verifiable encryption function.
 	VerifiableDecrypt(ciphertexts []types.Ciphertext) ([][]byte, error)
```

Comment:
This verifiable encrypt/decrypt now is incompatible with `Encrypt` and `Decrypt` though, isn't it ? 
Isn't this a bit problematic, in terms of the state we're leaving Dela in ? 🤔

Reply:
I see VerifiableEncrypt / VerifiableDecrypt working in pairs. Same for Encrypt / Decrypt / Reencrypt. You cannot use VerifiableEncrypt with Decrypt anyway.